src/python/pants/backend/experimental/jvm/resolve/coursier_fetch_integration_test.py

```diff
@@ -0,0 +1,306 @@
+                dependencies=MavenCoordinates([MavenCoord(coord="org.hamcrest:hamcrest-core:1.3")]),
+                file_digest=FileDigest(
+                    fingerprint="8e495b634469d64fb8acfa3495a065cbacc8a0fff55ce1e31007be4c16dc57d3",
+                    serialized_bytes_length=384581,
+                ),
+            )
+        ],
+    )
+    assert classpath_entry.coord == MavenCoord(coord="junit:junit:4.13.2")
+    assert classpath_entry.file_name == "junit-4.13.2.jar"
+    file_digest = rule_runner.request(
+        FileDigest, [ExtractFileDigest(classpath_entry.digest, "junit-4.13.2.jar")]
+    )
+    assert file_digest == FileDigest(
+        fingerprint="8e495b634469d64fb8acfa3495a065cbacc8a0fff55ce1e31007be4c16dc57d3",
+        serialized_bytes_length=384581,
+    )
+
+
+def test_fetch_one_coord_with_bad_fingerprint(rule_runner: RuleRunner) -> None:
+    # TODO: Use a mocked Process to test these kinds of details.  While this version hasn't changed
+    # in nearly a decade, it's still brittle in a unit test to rely on an externally fetched file staying stable.
+
+    expected_exception_msg = (
+        r".*?CoursierError:.*?Coursier fetch for .*?hamcrest.*? succeeded.*?"
+        r"66fdef91e9739348df7a096aa384a5685f4e875584cce89386a7a47251c4d8e9.*?"
+        r"did not match.*?ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+    )
+    with pytest.raises(ExecutionError, match=expected_exception_msg):
+        rule_runner.request(
+            ResolvedClasspathEntry,
+            [
+                CoursierLockfileEntry(
+                    coord=MavenCoord(coord="org.hamcrest:hamcrest-core:1.3"),
+                    file_name="hamcrest-core-1.3.jar",
+                    direct_dependencies=MavenCoordinates([]),
+                    dependencies=MavenCoordinates([]),
+                    file_digest=FileDigest(
+                        fingerprint="ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+                        serialized_bytes_length=45024,
+                    ),
+                )
+            ],
+        )
+
+
+def test_fetch_one_coord_with_bad_length(rule_runner: RuleRunner) -> None:
+    # TODO: Use a mocked Process to test these kinds of details.  While this version hasn't changed
+    # in nearly a decade, it's still brittle in a unit test to rely on an externally fetched file staying stable.
+    expected_exception_msg = (
+        r".*?CoursierError:.*?Coursier fetch for .*?hamcrest.*? succeeded.*?"
+        r"66fdef91e9739348df7a096aa384a5685f4e875584cce89386a7a47251c4d8e9.*?"
+        r"serialized_bytes_length=45024.*?"
+        r"did not match.*?66fdef91e9739348df7a096aa384a5685f4e875584cce89386a7a47251c4d8e9.*?"
+        r"serialized_bytes_length=1\).*?"
+    )
+    with pytest.raises(ExecutionError, match=expected_exception_msg):
+        rule_runner.request(
+            ResolvedClasspathEntry,
+            [
+                CoursierLockfileEntry(
```

Comment:
It is generally recommended to minimize the amount of code running under `pytest.raises` ideally just the code that will trigger the exception everything else should be before the `with` block (in this case, constructing the CoursierLockfileEntry object)
https://github.com/m-burst/flake8-pytest-style/blob/v1.4.1/docs/rules/PT012.md

Reply:
Done